### PR TITLE
Add 'LIVE' label to StoryPromo

### LINKED ISSUE #1356: Add 'LIVE' label to story promo for Live Page Promos
**Is your feature request related to a problem? Please describe.**
To enable LIV assets to be shown on our front pages, we need to be able to add a "LIVE" label before the headline text.

The label should not just be appended to the beginning of the headline, but rather be its own element because it's getting different typography.

**Describe the solution you'd like**
Add the ability to add a LIVE label to story promo. Maybe pass in a "isLive" prop, if yes, show the label. 

**Describe alternatives you've considered**
Honestly, no clue.
Making a separate component for the LIVE label, but that's just silly.

**Testing notes**
[Tester to complete]

Dev insight: Add a snapshot test to cover the scenario.

**Additional context**
WSFRONTPGS-81
[Designs + Accessibility](https://app.zeplin.io/project/5c8a5e021d454124ef90669f?seid=5d25bd8606494a2e258448cf
)

### PR DESCRIPTION
**Pull request by ghost**
Resolves #1356

**Overall change:** Add live label to `StoryPromo`

**Code changes:**

- Create `LiveLabel` component and export it.
- Add live story in storybook
- Add snapshot test for Live `StoryPromo`
- Update `README.MD`

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Comment by ghost** [comment]
A developer swarm needs to be done before this can be merged

File: packages/components/psammead-story-promo/src/index.jsx
Comment (ryanmccombe) [inline packages/components/psammead-story-promo/src/index.jsx]:
```suggestion
  dir: oneOf(['rtl', 'ltr']),
```

File: packages/components/psammead-story-promo/src/index.jsx
Comment (ryanmccombe) [inline packages/components/psammead-story-promo/src/index.jsx]: Guess we don't care about this hardcoded english yet?

**Review by ryanmccombe — APPROVED** [review]

File: packages/components/psammead-story-promo/src/index.jsx
Comment (ghost) [reply packages/components/psammead-story-promo/src/index.jsx]: We can add a `label` prop instead of hardcoding, so that we would be able to pass a translation if needed.

Comment (ryanmccombe) [reply packages/components/psammead-story-promo/src/index.jsx]: Sounds good :D

Comment (ghost) [reply packages/components/psammead-story-promo/src/index.jsx]: Or alternatively we could update it to receive children (`text node`)

Comment (ryanmccombe) [reply packages/components/psammead-story-promo/src/index.jsx]: Good point - `<LiveLabel>Blah</LiveLabel>` would probably be clearer than `<LiveLabel label="Blah" />` 👍

**Review by Bopchy — APPROVED** [review]

File: packages/components/psammead-story-promo/src/index.jsx
Comment (greenc05) [reply packages/components/psammead-story-promo/src/index.jsx]: I think this will always be in English... unless it's a RTL language

File: packages/components/psammead-story-promo/README.md
Comment (andrew-nowak) [inline packages/components/psammead-story-promo/README.md]: Not sure why this has to be wrapped in a span, couldn't this line just be `{headline}`?

File: packages/components/psammead-story-promo/src/index.test.jsx
Comment (andrew-nowak) [inline packages/components/psammead-story-promo/src/index.test.jsx]:
```suggestion
```
The headline should be in the `LiveComponent`

File: packages/components/psammead-story-promo/src/index.stories.jsx
Comment (andrew-nowak) [inline packages/components/psammead-story-promo/src/index.stories.jsx]: Could you add a story for a Live Top Story?

File: packages/components/psammead-story-promo/src/index.jsx
Comment (andrew-nowak) [inline packages/components/psammead-story-promo/src/index.jsx]: Is this needed?

File: packages/components/psammead-story-promo/package.json
Comment (DenisHdz) [inline packages/components/psammead-story-promo/package.json]:
```suggestion
    "@bbc/psammead-styles": "^1.2.0",
```

File: packages/components/psammead-story-promo/src/index.jsx
Comment (andrew-nowak) [inline packages/components/psammead-story-promo/src/index.jsx:263]: Is this needed?

**Review by DenisHdz — CHANGES_REQUESTED** [review]

**Review by andrew-nowak — CHANGES_REQUESTED** [review]

File: packages/components/psammead-story-promo/README.md
Comment (ghost) [reply packages/components/psammead-story-promo/README.md]: This is the expected markup
```html
<h3>
    <a href=”">
      <span role="text">
	    <span aria-hidden="true">LIVE</span>
	    <span class="vis-hidden" lang="en-GB">Live, </span>
	    <span>BBC Pidgin Minute</span>
	</span>
    </a>
</h3>
```
I have also sent you a document on slack.

File: packages/components/psammead-story-promo/src/index.jsx
Comment (ghost) [reply packages/components/psammead-story-promo/src/index.jsx]: This is not needed.

File: packages/components/psammead-story-promo/src/index.jsx
Comment (ghost) [reply packages/components/psammead-story-promo/src/index.jsx:263]: This is needed, if not we will end up with
Screenshot 2019-07-23 at 07 58 28
I confirmed with UX, the `LIVE` label should not be underlined on hover.

File: packages/components/psammead-story-promo/src/index.stories.jsx
Comment (ghost) [reply packages/components/psammead-story-promo/src/index.stories.jsx]: Added a knob to toggle the live status.

File: packages/components/psammead-story-promo/README.md
Comment (DenisHdz) [reply packages/components/psammead-story-promo/README.md]: I have seen @greenc05's comment in the issue, but in the `Navigation` we use the same kind of markup and we don't wrap the last element within a `span`. Could you confirm this with her?

```
 <Fragment>
    <StyledSpan role="text" script={script}>
      <VisuallyHiddenText>{currentPageText}, </VisuallyHiddenText>
      {link}
    </StyledSpan>
  </Fragment>
```

Comment (ghost) [reply packages/components/psammead-story-promo/README.md]: In
https://github.com/bbc/simorgh/blob/93c063c295a115ba0ae2c2900fa1c614040b5a50/src/app/containers/StoryPromo/LinkContents/index.jsx#L36-L43
we have a span wrapping the content. The question would be, does wrapping `headline` in a `<span>` while inside `<span aria-role='text'>` have any effect while using screen readers. if not then we can remove the wrapping. @greenc05

Comment (greenc05) [reply packages/components/psammead-story-promo/README.md]: In the above code example, the span around the text 'BBC Pidgin Minute' isn't needed from an accessibility point of view...

Comment (ghost) [reply packages/components/psammead-story-promo/README.md]: `<span>` wrapping removed. 👍

**Review by andrew-nowak — COMMENTED** [review]
Thanks for all the changes, this is looking really good! A couple of merge conflicts have appeared this morning but then it will lgtm

**Review by dr3 — APPROVED** [review]

**Review by j-pendlebury — APPROVED** [review]
Just needs conflicts to be resolved 👍

**Review by DenisHdz — APPROVED** [review]
Great work 👍🏻

**Review by andrew-nowak — APPROVED** [review]

**Comment by greenc05** [comment]
Just a reminder that this needs a developer accessibility swarm before it can be merged

**Comment by DenisHdz** [comment]
❓ = Not done
✅ = Done, pass
❌ = Done, fail
👀 = Review with UX

## Checklist - Manual ##

  - ### Design ✅

  - ### Color contrast  ✅

  - ### Color blindness ✅

  - ### High Contrast mode PC ✅ 

live-hc1
live-hc2

  - ### Color preferences changed in Firefox ✅ 
  
  - ### Text resized to 200% ✅

  - ### Page zoomed to 200% ✅

  - ### Heading order  ✅

  - ### Icons ❓

  - ### Keyboard navigation: tab order, focus css, trace tab path ✅

  - ### aXe  ✅

  - ### No CSS    ✅

  - ### No Javascript   ✅

  - ### bbc-a11y  ✅

## Checklist - Assistive Technology ##

Acceptance criteria:
```
Given I use a screen reader
When I navigate to a live promo
And I am navigating via headings and links
Then the live label should be announced as part of the heading/link
And the live label should be announced first

Given I use a screen reader (LTR languages)
When viewing a live promo
The ‘Live’ label should be read in English, with an English accent
And it should be pronounced as ‘live’ as in ‘live event’ rather than ‘live’ as in ‘I live here’
```

 - ###  VoiceOver OS (Mac) (Screen Reader) ✅

   > With Safari 12 on macOS Mojave

 - ### VoiceOver iOS (iPhone/iPad) (Screen Reader) ✅

   > With Safari 12 on iOS 12

 - ### TalkBack  (Screen Reader) ✅

   > With Chrome on Android 8 

    ~~When navigating via `Headings and Landmarks` the `Live` label is not announced, as it does when navigating via `Links` or `Default`.~~

- ### ZoomText (Screen Magnifier with Screen Reader capabilities) ✅

   > With IE11 on Windows 7

- ### JAWS  (Screen Reader) ✅

   > With IE11 on Windows 7

- ### Read&Write  (Reading Solution) ✅

   > With IE11 on Windows 7

  ~~It also reads out the  aria-hidden `LiveLabel`: `live live`~~

- ### NVDA  (Screen Reader) ✅

   > With IE11 on Windows 7

  ~~It also reads out the  aria-hidden `LiveLabel`: `live live`~~

**Comment by WeimingFan** [comment]
UX Check: CSS is fine. Fan

**Comment by DenisHdz** [comment]
This is ready for Test 🎉

The few bugs found during the swarm, were due to be testing the component in the Storybook UI iframe and not opening the canvas in a new tab.

The reason why it was initially tested like that, was because the `Live label` is added using a `knob` and in previous versions of `@storybook/addon-knobs`, all the options changed via `knobs` disappear when opening in it a new tab.

**Comment by paruchurisilpa** [comment]
Looks good to me..

**Pull request merged**
